legacy/application/models/ShowInstance.php

```diff
@@ -844,4 +844,33 @@ public function isRepeating()
     {
         return $this->getShow()->isRepeating();
     }
+
+    public function trimOverbooked()
+    {
+        // iterate through all of the showinstance items and delete those with start time greater than endtime of the show
+        $showEnd = new DateTime($this->getShowInstanceEnd(), new DateTimeZone('UTC'));
+        foreach ($this->getShowListContent('UTC') as $item) {
+            $itemStart = new DateTime($item['starts'], new DateTimeZone('UTC'));
+            if ($itemStart->getTimestamp() > $showEnd->getTimestamp()) {
+                $this->removeItem($item['item_id'], $item['starts']);
+            }
+        }
+    }
```

Comment:
Would it be possible to execute all the remove Item at once? This feels like a lot for sql requests, or not?

Reply:
I'm not sure how to make a combined sql call delete the entry that match exactly the listed tuples, so i opted for the loop approach. I'll look into the table definition if there is a primary key that can be used instead.

Reply:
If we want to rely on the playout status (seem risky, don't know if this will break something else down the line), we can do: 

```sql
      DELETE FROM cc_schedule
           WHERE playout_status == 2 -- please make sure 2 is the right status.
        SQL;
```

But maybe a single explicit sql query can handle this.

Reply:
playout_status was inconsistent in my live db so i didn't use it. there's a primary key we can use instead.